crates/bevy_render/src/render_resource/storage_buffer.rs

```diff
@@ -60,14 +64,19 @@ impl<T: ShaderType + WriteInto> StorageBuffer<T> {
         &mut self.value
     }
 
+    /// Debug label of a buffer. This will show up in graphics debuggers for easy identification.
+    pub fn label(&mut self, label: Option<Cow<'static, str>>) {
```

Comment:
I suppose if the label is changed, then the only way that would become visible would be for the buffer to be reallocated. That's a shame, but it's the reality. As such, I think we need to implement `get_label()`, and `set_label()`. In `set_label()`, check if the label changed, and if so, mark that for the next `reserve()` call, the buffer will need to be reallocated regardless of capacity. Then when reserve() is called, if the label changed bool is set, reset it to false, and reallocate the buffer with the new label. I'm suggesting doing it this way so we don't reallocate the buffer at the wrong time and break everything.

Also, I'm pretty sure `wgpu` accepts `Some(&str)` as we use things like `Some("main_opaque_pass_3d")` and that's a much nicer API than `Cow`.

Reply:
Is there a reason ComputePipelineDescriptor stores its label as a Cow<'static, str> ?

Reply:
I suspect this was a misguided performance optimization. The `Cow` struct doesn't work the way that @cart thought it did originally.